### PR TITLE
fix: nil pointer on create instance add machine

### PR DESCRIPTION
**Pull request by stebenz**
### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Review by muhlemmer — DISMISSED** [review]

File: cmd/defaults.yaml [hunk 1]
```diff
@@ -433,6 +433,7 @@ DefaultInstance:
         Type:
       Pat:
         ExpirationDate:
+        Scopes:
```
Comment (muhlemmer) [inline cmd/defaults.yaml:436]: I'm not sure about this added option. I don't see anywhere else in the PR code. Probably I'm missing something..

File: internal/api/grpc/system/instance_converter.go [hunk 2]
```diff
@@ -114,9 +114,9 @@ func createInstancePbToAddMachine(req *system_pb.CreateInstanceRequest_Machine,
 			Scopes: []string{oidc.ScopeOpenID, z_oidc.ScopeUserMetaData, z_oidc.ScopeResourceOwner},
 		}
 
-		if !defaultMachine.Pat.ExpirationDate.IsZero() {
+		if defaultMachine.Pat != nil && !defaultMachine.Pat.ExpirationDate.IsZero() {
 			pat.ExpirationDate = defaultMachine.Pat.ExpirationDate
-		} else if req.PersonalAccessToken.ExpirationDate.IsValid() {
+		} else if req.PersonalAccessToken != nil && req.PersonalAccessToken.ExpirationDate.IsValid() {
```
Comment (muhlemmer) [inline internal/api/grpc/system/instance_converter.go:119]: Generated proto messages provide nil-safe getters:

```suggestion
		} else if req.GetPersonalAccessToken().GetExpirationDate().IsValid() {
```

**Review by muhlemmer — DISMISSED** [review]
Left some comments

File: cmd/setup/03.go [hunk 3]
```diff
@@ -101,25 +111,41 @@ func (mig *FirstInstance) Execute(ctx context.Context) error {
...
-	f := os.Stdout
-	if mig.MachineKeyPath != "" {
+	if mig.MachineKeyPath != "" && key != nil {
+		f := os.Stdout
+		defer f.Close()
 		f, err = os.OpenFile(mig.MachineKeyPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
 		if err != nil {
 			return err
 		}
-		defer f.Close()
```
[the remark this answers is not in the document]
Comment (stebenz) [reply cmd/setup/03.go:115]: separate function, handles output to stdout or file

File: cmd/setup/03.go [hunk 4]
```diff
@@ -101,25 +111,41 @@ func (mig *FirstInstance) Execute(ctx context.Context) error {
...
+
+		keyDetails, err := key.Detail()
+		if err != nil {
+			return err
+		}
+		_, err = fmt.Fprintln(f, string(keyDetails))
```
[the remark this answers is not in the document]
Comment (stebenz) [reply cmd/setup/03.go]: error gets returned now

File: cmd/defaults.yaml (hunk 1 above)
Comment (stebenz) [reply cmd/defaults.yaml]: deleted scope everywhere as it should not be configurable

**Review by muhlemmer — APPROVED** [review]

**Pull request merged**